File: tm/NativeCalculatorAppRust.ts
/* eslint-disable prettier/prettier */
/* eslint-disable semi */
import { TurboModule, TurboModuleRegistry } from 'react-native'

export interface Spec extends TurboModule {
    readonly add: (x: number, y: number) => number;
    readonly sub: (x: number, y: number) => number;
    readonly mul: (x: number, y: number) => number;
    readonly div: (x: number, y: number) => number;
}

export default TurboModuleRegistry.getEnforcing<Spec>(
    'NativeCalculatorAppRust'
)

File: tm/NativeCalculatorAppRust.cpp
#include "NativeCalculatorAppRust.h"

namespace facebook::react {
NativeCalculatorAppRust::NativeCalculatorAppRust(
    std::shared_ptr<CallInvoker> jsInvoker)
    : NativeCalculatorAppRustCxxSpec(std::move(jsInvoker)) {}

int NativeCalculatorAppRust::add(jsi::Runtime &rt, int x, int y) {
  return x + y;
}
int NativeCalculatorAppRust::sub(jsi::Runtime &rt, int x, int y) {
  return x - y;
}
int NativeCalculatorAppRust::mul(jsi::Runtime &rt, int x, int y) {
  return x * y;
}
int NativeCalculatorAppRust::div(jsi::Runtime &rt, int x, int y) {
  if (y == 0) {
    return 0;
  }
  return x / y;
}
} // namespace facebook::react